mlir/lib/Dialect/MIOpen/Transforms/GridwiseGemmToBlockwise.cpp

```diff
@@ -2512,32 +2516,50 @@ struct GridwiseGemmV2RewritePattern
 
     if (enableOutSwizzles) {
       Value laneId = b.create<arith::RemUIOp>(loc, tid, waveSizeConstantOp);
-      for (Value result : tailResults) {
+      for (int i = 0; i < vectorNumber; ++i) {
+        Value indexOp = b.createOrFold<ConstantIndexOp>(loc, i);
+        Value loaded =
+            b.create<memref::LoadOp>(loc, vectorType, regCAllocOp, indexOp);
         Value swizzle = b.create<InWarpTransposeOp>(
-            loc, result.getType(), result, laneId,
-            b.getI32IntegerAttr(group_size), b.getI32ArrayAttr({0, 1, 2, 3}));
+            loc, vectorType, loaded, laneId, b.getI32IntegerAttr(group_size),
+            b.getI32ArrayAttr({0, 1, 2, 3}));
         transformedTail.push_back(swizzle);
+        b.create<memref::StoreOp>(loc, swizzle, regCAllocOp, indexOp);
       }
     } else {
       llvm::copy(tailResults, std::back_inserter(transformedTail));
     }
 
-    // Convert GEMM results to the expected output type (so we can fuse in)
-    // operations expecting that type before writeback and store
-    // the result vectors into a allocation of registers to maintain uniformity
-    // with the non-xdlops gemm. (These "stores" will be optimized out)
-    Type destType = op.c().getType().cast<MemRefType>().getElementType();
-    MemRefType mergedType = MemRefType::get(
-        numElements, destType, {},
+    Value registerC = regCAllocOp;
+    auto convertedCType = MemRefType::get(
+        vectorType.getNumElements() * vectorNumber, destType, {},
         /*memorySpace=*/gpu::GPUDialect::getPrivateAddressSpace());
-    VectorType castVectorType = vectorType.clone(destType);
-    Value resultMerged = b.create<miopen::GpuAllocOp>(loc, mergedType);
-    for (const auto &pair : llvm::enumerate(transformedTail)) {
-      Value cast = createTypeConversionOp(b, loc, pair.value(), castVectorType);
-      Value offset = b.createOrFold<arith::ConstantIndexOp>(
-          loc, pair.index() * resultCVectorLen);
-      b.create<miopen::InBoundsStoreOp>(loc, cast, resultMerged, offset);
+    Value convertedC = b.create<miopen::GpuAllocOp>(loc, convertedCType);
+
+    // Convert from memref<?xvector<?xT>> to memref<?xT> where the source T
+    // is the accumulatorType and destination type is destType
+    auto convertLoop = b.create<TransformingForOp>(
```

Comment:
Instead of the `MulIOp`, can we have the `bounds` be the size of the destination memref and the `strides` be the size of the vector?

Reply:
Nice suggestion! I don't like this `MulIOp` to start with, I'm glad you point this out to me, thank you!

Reply:
Wait, how does it work to make the source and destination in two coordinate space?

This transforming for op would:
 - `load registerC[coord]`
 - `type convert`
 - `store loaded convertedC[coord * vectorLen]` 

What should I do such that the load on registerC is still on linear coordinate space whereas store is strided by `vectorLen`

Reply:
Good question. Earlier idea withdrawn. My gut is saying to make a quick map that `Embed`s the destination coordinate with the appropriate coefficient to scale it from `[0, 1, ...]` to `[0, L, 2L, ...]` so that the multiplication is part of the loop structure, and thus analyzable.

Reply:
To rephrase, instead of having one iteration domain, have two
The second one is
```
int64_t numOutputs = vectorType.getNumElements() * numVectors;
BottomUpTMBuilder vectorToScalar(b, {"scalar"}, {numOutputs}, loc);
vectorDest.embed({"vector"}, {0}, {numVectors}, "vector", {vectorType.getNumElements()});
TransformMapAttr vectorToScalarAttr = vectorToScalar.get();

...
// Set up the TransformingFor to take both the no-transforms array (vector coordinates) and the array containing the above transform (which gives scalar coordinates)
```

Reply:
Addressed in a7f8be7